build-tools-internal/src/main/groovy/org/elasticsearch/gradle/internal/doc/SnippetsTask.groovy

```diff
@@ -304,7 +304,7 @@ class SnippetsTask extends DefaultTask {
     }
 
     static Source matchSource(String line) {
-        def matcher = line =~ /\["?source"?,\s*"?([-\w]+)"?(,((?!id=).)*(id="?([-\w]+)"?)?(.*))?].*/
+        def matcher = line =~ /\["?source"?(?:\.[^,]+)?,\s*"?([-\w]+)"?(,((?!id=).)*(id="?([-\w]+)"?)?(.*))?].*/
```

Comment:
I just trust your regex skills here

Reply:
It a new optional, non-capturing group that matching `.` followed by anything but a `,`.